pgsql/bin/functions/system_stop

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+SIGNAL="$1"
+
+if [ -f "$PGDATA/postmaster.pid" ]; then
+    # How to stop postgres
+    # https://www.postgresql.org/docs/9.6/static/server-shutdown.html
+    case "$SIGNAL" in
+        TERM|SIGTERM) # Smart
+            echo ">>>>>> Stopping postgres in Smart mode..."
+            gosu postgres pg_ctl stop -m smart
```

Comment:
then you should use `SIGINT` to stop your container... my be I did not get your problem, but docker-compose allows to set stop signal, hopefully kubernetes as well.